browser/ui/webui/settings/brave_importer_observer.cc

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/settings/brave_importer_observer.h"
+
+#include <utility>
+
+#include "base/logging.h"
+#include "chrome/browser/importer/external_process_importer_host.h"
+
+namespace settings {
+
+BraveImporterObserver::BraveImporterObserver(
+    ExternalProcessImporterHost* importer_host,
+    const importer::SourceProfile& source_profile,
+    uint16_t imported_items,
+    ReportProgressCallback callback)
+    : source_profile_(source_profile),
+      imported_items_(imported_items),
+      callback_(std::move(callback)),
+      importer_host_(importer_host) {
+  DCHECK(importer_host);
+}
+
+BraveImporterObserver::~BraveImporterObserver() {
```

Comment:
Q) When this observer is destroyed? I can't find when it does.

Reply:
Hmm, I thought it would be better to destroy each observer when each importing is ended from the resource management point of view instead of destroying all when welcome page is closed.
If you think it's trivial, I'm fine with current PR.

Reply:
`ImportEnded` is called inside the observer, I want to avoid destroying itself during execution.

Reply:
I think `BraveImportDataHandler` can destroy it from `BraveImportDataHandler::NotifyImportProgress()`
by checking value's event key - `"event", "ImportEnded"`. Is this too tricky?

Reply:
done as discussed in Slack